### PR TITLE
Query - Fix offset and post per page in preview

### LINKED ISSUE #1448: Add offset parameter to post meta query type (GB 2.0)
### Description
It would be useful to have an offset parameter for post meta query type (relationship type custom field)

#### Use case:
For example, the user has 3 related posts in a relationship field, and he wants the 1st post to show after the 3rd h2, the 2nd post to show after the 6th h2, and the 3rd one after the 10th h2, it's very common to have related posts across the entire post content.

With offset, we can add 3 query blocks across the post content.

Thank you!

-----

### PR DESCRIPTION
**Pull request by iansvo**
Fixes #1448 
Requires: https://github.com/tomusborne/generateblocks-pro/pull/769

### CODE REVIEW

File: src/blocks/looper/components/LoopInnerBlocksRenderer.jsx [hunk 1]
```diff
@@ -235,18 +235,25 @@ export function LoopInnerBlocksRenderer( props ) {
...
-				? query?.posts_per_page
-				: 10;
+			let { posts_per_page: perPage = 10, offset = 0 } = query;
 
-			if ( '-1' === perPage?.toString() ) {
+			// Ensure the params are a valid integer for comparison.
+			perPage = parseInt( perPage, 10 );
+			offset = parseInt( offset, 10 );
+
+			console.log( { perPage, offset } );
```
Comment (tomusborne) [inline src/blocks/looper/components/LoopInnerBlocksRenderer.jsx:244]: Need to remove this.

**Review by tomusborne — CHANGES_REQUESTED** [review]

File: src/blocks/looper/components/LoopInnerBlocksRenderer.jsx (hunk 1 above)
Comment (iansvo) [reply src/blocks/looper/components/LoopInnerBlocksRenderer.jsx]: Done.

**Comment by iansvo** [comment]
@tomusborne As noted on the free PR, I had to make some adjustments to the way the pagination works to properly support non post-type queries. 

I confirmed it works as expected on frontend and in preview with all three query types.

File: includes/blocks/class-query-page-numbers.php [hunk 2]
```diff
@@ -39,19 +39,20 @@ public static function render_block( $attributes, $block_content, $block ) {
...
-			$is_wp_query = GenerateBlocks_Block_Query::TYPE_WP_QUERY === $query_type;
+			$query_data = $block->context['generateblocks/queryData'] ?? null;
+			$max_pages  = $block->context['generateblocks/maxPages'] ?? 0;
 
-			if ( ! $query_data || ( ! $is_wp_query && ! is_array( $query_data ) ) ) {
+			if ( ! $query_data ) {
 				return '';
 			}
 
+			$query_data;
```
Comment (tomusborne) [inline includes/blocks/class-query-page-numbers.php:49]: What's this guy doing here?

**Pull request merged**